suc/pkg/service/service.go

```diff
@@ -1,86 +1,132 @@
+// If the provided service does not exist, a nil error and a false boolean will be returned.
+// The caller of Open is responsible for closing the returned Service (via Service.Close()).
+func Open(name string) (service *Service, serviceExists bool, err error) {
+	logrus.Debugf("Opening %s service", name)
+	svcMgr, err := mgr.Connect()
 	if err != nil {
-		logrus.Errorf("Cannot restart %s as the service failed to open: %v", defaults.WindowsServiceName, err)
-		return fmt.Errorf("failed to refresh the %s service: %w", winSrv.Name, err)
+		return nil, false, fmt.Errorf("failed to connect to service manager: %w", err)
 	}
-	defer winSrv.Close()
+	defer svcMgr.Disconnect()
 
-	err = restartService(winSrv)
+	s, err := svcMgr.OpenService(name)
+	doesNotExist := errors.Is(err, windows.ERROR_SERVICE_DOES_NOT_EXIST)
+	if err != nil && !doesNotExist {
+		return nil, false, fmt.Errorf("failed to open service %s via service manager: %w", name, err)
+	}
+
+	if doesNotExist {
+		return nil, false, nil
+	}
+
+	if s == nil {
+		return nil, false, fmt.Errorf("failed to open service %s, a nil service was returned", name)
+	}
+
+	cfg, err := s.Config()
 	if err != nil {
-		return fmt.Errorf("failed to refresh the %s service: %w", winSrv.Name, err)
+		return nil, false, fmt.Errorf("failed to open config for service %s via service manager: %w", name, err)
 	}
 
-	return nil
+	return &Service{
+		Name:   name,
+		svc:    s,
+		Config: cfg,
+	}, true, nil
 }
 
-func getServiceState(s *mgr.Service) (svc.State, error) {
-	q, err := s.Query()
+// GetState queries the Service and returns the svc.State
+func (s *Service) GetState() (svc.State, error) {
+	q, err := s.svc.Query()
 	if err != nil {
 		return 0, fmt.Errorf("could not query service %s: %w", s.Name, err)
 	}
 
 	return q.State, nil
 }
 
-func restartService(s *mgr.Service) error {
-	currentState, err := getServiceState(s)
+// Restart explicitly stops and then starts the Service.
+// Restart blocks for up to 25 seconds, or until the svc.State transitions to svc.Running
+func (s *Service) Restart() error {
+	logrus.Infof("Restarting %s service", s.Name)
+	currentState, err := s.GetState()
 	if err != nil {
 		return fmt.Errorf("failed to get state of service %s: %w", s.Name, err)
 	}
 
 	if currentState == svc.Running {
-		if err = stopService(s); err != nil {
+		if err = s.Stop(); err != nil {
 			logrus.Errorf("Encountered error attempting to stop the %s service: %v", s.Name, err)
 			return fmt.Errorf("encountered error attempting to stop the %s service: %w", s.Name, err)
 		}
 	}
 
-	if err = s.Start(); err != nil {
-		return fmt.Errorf("failed to restart the %s service: %w", s.Name, err)
+	if err = s.svc.Start(); err != nil {
+		return fmt.Errorf("failed to start the %s service while attempting to restart: %w", s.Name, err)
 	}
 
-	return waitForServiceState(s, svc.Running, 5, 5)
+	return s.WaitForState(svc.Running, 5, 5)
```

Comment:
As the same values for `delayInSeconds` and `maxAttempts` are used at multiple places, it is better to define the values as package-level constants.